xmtp_mls/src/groups/message_history.rs

```diff
@@ -265,7 +265,7 @@ where
             })?;
 
         // publish the intent
-        if let Err(err) = sync_group.publish_intents(&conn.into(), self).await {
+        if let Err(err) = sync_group.publish_messages(self).await {
```

Comment:
`publish_intents` will stop at the first commit - so if a key rotation or an add_missing_installations is queued up first, the message history reply won't actually be sent. Instead, use `publish_messages` after a `prepare_message` call, just like apps do.